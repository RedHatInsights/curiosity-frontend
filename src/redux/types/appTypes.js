/**
 * @memberof Types
 * @module AppTypes
 */

const STATUS_4XX = '4XX';
const STATUS_5XX = '5XX';
const SET_PRODUCT = 'SET_PRODUCT';
const SET_PRODUCT_VARIANT = 'SET_PRODUCT_VARIANT';
const SET_PRODUCT_VARIANT_QUERY_RESET_ALL = 'SET_PRODUCT_VARIANT_QUERY_RESET_ALL';

/**
 * Application action, reducer types.
 *
 * @type {{STATUS_4XX: string, SET_PRODUCT_VARIANT: string, SET_PRODUCT_VARIANT_QUERY_RESET_ALL: string,
 *     SET_PRODUCT: string, STATUS_5XX: string}}
 */
const appTypes = {
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};

export {
  appTypes as default,
  appTypes,
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};
