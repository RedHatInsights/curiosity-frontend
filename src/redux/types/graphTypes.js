/**
 * @memberof Types
 * @module GraphTypes
 */

const SET_GRAPH_LEGEND = 'SET_GRAPH_LEGEND';

/**
 * Graph action, reducer types.
 *
 * @type {{SET_GRAPH_LEGEND: string}}
 */
const graphTypes = {
  SET_GRAPH_LEGEND
};

export { graphTypes as default, graphTypes, SET_GRAPH_LEGEND };
