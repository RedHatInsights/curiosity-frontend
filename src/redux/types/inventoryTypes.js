const SET_INVENTORY_TAB = 'SET_INVENTORY_TAB';

/**
 * Inventory action, reducer types.
 *
 * @type {{SET_INVENTORY_TAB: string}}
 */
const inventoryTypes = {
  SET_INVENTORY_TAB
};

export { inventoryTypes as default, inventoryTypes, SET_INVENTORY_TAB };
