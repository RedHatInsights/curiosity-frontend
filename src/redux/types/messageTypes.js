/**
 * @memberof Types
 * @module MessageTypes
 */

const SET_BANNER_MESSAGES = 'SET_BANNER_MESSAGES';

/**
 * User action, reducer types.
 *
 * @type {{SET_BANNER_MESSAGES: string}}
 */
const messageTypes = { SET_BANNER_MESSAGES };

export { messageTypes as default, messageTypes, SET_BANNER_MESSAGES };
