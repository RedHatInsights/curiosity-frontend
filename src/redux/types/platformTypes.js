const PLATFORM_INIT = 'PLATFORM_INIT';
const PLATFORM_APP_NAME = 'PLATFORM_APP_NAME';
const PLATFORM_ON_NAV = 'PLATFORM_ON_NAV';
const PLATFORM_SET_NAV = 'PLATFORM_SET_NAV';

const platformTypes = { PLATFORM_INIT, PLATFORM_APP_NAME, PLATFORM_ON_NAV, PLATFORM_SET_NAV };

export { platformTypes as default, platformTypes, PLATFORM_INIT, PLATFORM_APP_NAME, PLATFORM_ON_NAV, PLATFORM_SET_NAV };
