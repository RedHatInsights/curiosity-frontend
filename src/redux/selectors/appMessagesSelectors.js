import { createSelector } from 'reselect';
import { rhsmApiTypes } from '../../types';
/**
 * Selector cache.
 *
 * @private
 * @type {{data: {object}}}
 */
const selectorCache = { data: {} };

/**
 * Return a combined state, props object.
 *
 * @private
 * @param {object} state
 * @param {object} props
 * @returns {object}
 */
const statePropsFilter = (state, props = {}) => ({
  reportCapacity: state.graph?.reportCapacity?.[props.productId],
  viewId: props.viewId,
  productId: props.productId
});

/**
 * Create selector, transform combined state, props into a consumable object.
 *
 * @type {{appMessages: {cloudigradeMismatch: boolean}}}
 */
const selector = createSelector([statePropsFilter], data => {
  const { viewId = null, productId = null, reportCapacity = {} } = data || {};
  const appMessages = {
    cloudigradeMismatch: false
  };

  const cache = (viewId && productId && selectorCache.data[`${viewId}_${productId}`]) || undefined;

  Object.assign(appMessages, { ...cache });

  // Scan Tally response for Cloud Meter flags
  if (reportCapacity.fulfilled && appMessages.cloudigradeMismatch !== true) {
    const [{ [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: reportData = [] }] = reportCapacity.data || {};

    const cloudigradeMismatch = reportData.find(
      ({ [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: mismatch }) => mismatch === true
    );

    appMessages.cloudigradeMismatch = cloudigradeMismatch !== undefined;

    selectorCache.data[`${viewId}_${productId}`] = {
      ...appMessages
    };
  }

  return { appMessages };
});

/**
 * Expose selector instance. For scenarios where a selector is reused across component instances.
 *
 * @param {object} defaultProps
 * @returns {{appMessages: {cloudigradeMismatch: boolean}}}
 */
const makeSelector = defaultProps => (state, props) => ({
  ...selector(state, props, defaultProps)
});

const appMessagesSelectors = {
  appMessages: selector,
  makeAppMessages: makeSelector
};

export { appMessagesSelectors as default, appMessagesSelectors, selector, makeSelector };
