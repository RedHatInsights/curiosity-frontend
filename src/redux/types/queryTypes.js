import { RHSM_API_QUERY_TYPES } from '../../types/rhsmApiTypes';

const SET_QUERY_CLEAR = 'SET_QUERY_CLEAR';
const SET_QUERY_CLEAR_INVENTORY_LIST = 'SET_QUERY_CLEAR_INVENTORY_LIST';

const SET_QUERY_RHSM_TYPES = {
  [RHSM_API_QUERY_TYPES.GRANULARITY]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.GRANULARITY}`,
  [RHSM_API_QUERY_TYPES.SLA]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.SLA}`,
  [RHSM_API_QUERY_TYPES.UOM]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.UOM}`,
  [RHSM_API_QUERY_TYPES.USAGE]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.USAGE}`
};

const SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES = {
  [RHSM_API_QUERY_TYPES.DIRECTION]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};

const SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES = {
  [RHSM_API_QUERY_TYPES.DIRECTION]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};

/**
 * Query/filter reducer types.
 *
 * @type {{SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES: string, SET_QUERY_RHSM_TYPES: string,
 *     SET_QUERY_CLEAR: string, SET_QUERY_CLEAR_INVENTORY_LIST: string,
 *     SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES: string}}
 */
const queryTypes = {
  SET_QUERY_CLEAR,
  SET_QUERY_CLEAR_INVENTORY_LIST,
  SET_QUERY_RHSM_TYPES,
  SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES,
  SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES
};

export {
  queryTypes as default,
  queryTypes,
  SET_QUERY_CLEAR,
  SET_QUERY_CLEAR_INVENTORY_LIST,
  SET_QUERY_RHSM_TYPES,
  SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES,
  SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES
};
