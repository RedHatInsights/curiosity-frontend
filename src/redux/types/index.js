import { appTypes } from './appTypes';
import { graphTypes } from './graphTypes';
import { platformTypes } from './platformTypes';
import { queryTypes } from './queryTypes';
import { rhsmTypes } from './rhsmTypes';
import { toolbarTypes } from './toolbarTypes';
import { userTypes } from './userTypes';

const reduxTypes = {
  app: appTypes,
  graph: graphTypes,
  platform: platformTypes,
  query: queryTypes,
  rhsm: rhsmTypes,
  toolbar: toolbarTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  appTypes,
  graphTypes,
  platformTypes,
  queryTypes,
  rhsmTypes,
  toolbarTypes,
  userTypes
};
