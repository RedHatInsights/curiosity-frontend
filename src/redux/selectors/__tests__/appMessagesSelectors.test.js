import appMessagesSelectors from '../appMessagesSelectors';
import { rhsmApiTypes } from '../../../types/rhsmApiTypes';

describe('AppMessagesSelectors', () => {
  it('should return specific selectors', () => {
    expect(appMessagesSelectors).toMatchSnapshot('selectors');
  });

  it('should pass minimal data on missing a reducer response', () => {
    const state = {};
    expect(appMessagesSelectors.appMessages(state)).toMatchSnapshot('missing reducer error');
  });

  it('should pass minimal data on a product ID without a product ID provided', () => {
    const props = {
      viewId: 'test',
      productId: undefined
    };
    const state = {
      messages: {
        report: {
          fulfilled: true,
          data: { [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: [] }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(state, props)).toMatchSnapshot('no product id error');
  });

  it('should populate data on a product ID when the api response is missing expected properties', () => {
    const props = {
      viewId: 'test',
      productId: 'Lorem Ipsum missing expected properties'
    };
    const state = {
      messages: {
        report: {
          'Lorem Ipsum missing expected properties': {
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-04T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 1
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-05T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 1
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-06T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 4,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 2
                }
              ]
            }
          }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(state, props)).toMatchSnapshot('data populated, missing properties');
  });

  it('should map a fulfilled product ID response to an aggregated output', () => {
    const props = {
      viewId: 'test',
      productId: 'Lorem Ipsum fulfilled aggregated output'
    };
    const state = {
      messages: {
        report: {
          'Lorem Ipsum fulfilled aggregated output': {
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-04T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: true
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-05T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: true
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-06T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.CORES]: 4,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 4,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: null,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: true
                }
              ]
            }
          }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(state, props)).toMatchSnapshot('fulfilled');
  });

  it('should populate data from the in memory cache', () => {
    const props = {
      viewId: 'cache-test',
      productId: 'Lorem Ipsum ID cached'
    };
    const stateInitialFulfilled = {
      messages: {
        report: {
          'Lorem Ipsum ID cached': {
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2019-09-04T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: true
                }
              ]
            }
          }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(stateInitialFulfilled, props)).toMatchSnapshot(
      'cached data: initial fulfilled'
    );

    const statePending = {
      messages: {
        report: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.messages.report['Lorem Ipsum ID cached'],
            pending: true
          }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(statePending, props)).toMatchSnapshot(
      'cached data: cache used and pending'
    );

    const stateFulfilled = {
      messages: {
        report: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.messages.report['Lorem Ipsum ID cached'],
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.DATE]: '2018-07-04T00:00:00.000Z',
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.CORES]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.SOCKETS]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HYPERVISOR_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_CORES]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.PHYSICAL_SOCKETS]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_DATA]: true,
                  [rhsmApiTypes.RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES.HAS_CLOUDIGRADE_MISMATCH]: false
                }
              ]
            }
          }
        }
      }
    };

    expect(appMessagesSelectors.appMessages(stateFulfilled, props)).toMatchSnapshot(
      'cached data: update and fulfilled'
    );
  });
});
