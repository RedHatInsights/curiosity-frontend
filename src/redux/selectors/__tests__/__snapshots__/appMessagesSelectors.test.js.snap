// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMessagesSelectors should pass minimal data on a product ID without a product ID provided: no product id error 1`] = `
Object {
  "appMessages": Object {},
}
`;

exports[`AppMessagesSelectors should pass minimal data on missing a reducer response: missing reducer error 1`] = `
Object {
  "appMessages": Object {},
}
`;

exports[`AppMessagesSelectors should return specific selectors: selectors 1`] = `
Object {
  "appMessages": [Function],
  "makeAppMessages": [Function],
}
`;
